addons/mrp/models/mrp_workorder.py

```diff
@@ -679,7 +679,14 @@ def button_finish(self):
                 vals['date_start'] = end_date
             if not workorder.date_planned_start or end_date < workorder.date_planned_start:
                 vals['date_planned_start'] = end_date
+            manual_duration_expected = workorder.duration_expected != workorder._get_duration_expected()
             workorder.with_context(bypass_duration_calculation=True).write(vals)
+            # Update the duration_expected for done workorders as they are
+            # no more updated by the `_compute_duration_expected`
+            if not manual_duration_expected:
+                workorder.duration_expected = workorder._get_duration_expected()
+            if not workorder.duration:
+                workorder.duration = workorder.duration_expected
```

Comment:
this doesn't really make sense when reading it:
- let's say current `duration_expected` is 30 min and `_get_duration_expected` also returns 30 min
- this means `manual_duration_expected` is False
- when reaching `if not manual_duration_expected:`, we get inside (because `not False == True`)
- we replace 30 min (`duration_expected`) by 30 min (`_get_duration_expected`)

So yeah, not sure what's the point of this. Is `duration_expected` rewritten in the `workorder.with_context(bypass_duration_calculation=True).write(vals)`? Also, you can assign `_get_duration_expected` to a variable to only call it once at the beginning, for optimization.

Reply:
You are totally right, this part of the fix is useless.This is a left over of a prior fix attempt that I forgot to remove after I decided to make the change with `qty_producing` in the onchange. My changes used to take care of `qty_produced` and not the `qty_producing` so that the `_get_duration_expected` was changing after validation of the workorders taking care of the real quantity produced (however the change was not visible on change on the form so this solution was discarded). Sorry for that !